src/components/forms/RoutingTableRouteForm.js

```diff
@@ -0,0 +1,134 @@
+import React, { Component } from "react";
+import { buildFormFunctions, buildFormDefaultInputMethods } from "../component-utils";
+import { IcseNameInput, IcseTextInput } from "../Inputs";
+import { IcseSelect, IcseNumberSelect } from "../Dropdowns";
+import { IcseFormGroup } from "../Utils"
+import PropTypes from "prop-types";
+import { isNullOrEmptyString, isIpv4CidrOrAddress } from "lazy-z";
+
+class RoutingTableRouteForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { ...this.props.data };
+    if (
+      !isNullOrEmptyString(this.state.action) &&
+      this.state.action !== "deliver"
+    ) {
+      this.state.next_hop = "0.0.0.0";
+    }
+    this.handleInputChange = this.handleInputChange.bind(this);
+    buildFormFunctions(this);
+    buildFormDefaultInputMethods(this);
+  }
+
+  /**
+   * handle input change
+   * @param {string} name key to change in state
+   * @param {*} value value to update
+   */
+  handleInputChange(event) {
+    let nextState = { ...this.state };
+    let { name, value } = event.target;
+    nextState[name] = value;
+    if (name === "action" && value !== "deliver") {
+      nextState.next_hop = "0.0.0.0";
+    } else if (name === "action") {
+      nextState.next_hop = null;
+    }
+    this.setState(nextState);
+  }
+
+  render() {
+    return (
+      <>
+        <IcseFormGroup>
+          <IcseNameInput
+            componentName="routing-table-route"
+            id={this.props.data.name + "-route-name"}
+            hideHelperText
+            value={this.state.name}
+            onChange={this.handleInputChange}
+            invalidCallback={() => {
+              this.props.invalidCallback(this.state, this.props);
+            }}
```

Comment:
same as above, should be 
```js
invalid={this.props.invalidCallback(this.state, this.props}
```

Reply:
this was not working for me, needed to return a function

Reply:
nm